test/unit/format/vcf_input_test.cpp

```diff
@@ -0,0 +1,211 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// Copyright (c) 2020-2021, deCODE Genetics
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <bio/format/vcf_input_handler.hpp>
+#include <bio/var_io/reader.hpp>
+#include <seqan3/core/debug_stream.hpp>
+#include <seqan3/test/expect_range_eq.hpp>
+
+#include "vcf_data.hpp"
+
+template <bio::ownership own>
+void field_types_bcf_style()
+{
+    std::istringstream istr{std::string{example_from_spec}};
+
+    using record_t =
+      bio::detail::record_from_typelist<std::remove_cvref_t<decltype(bio::var_io::default_field_ids)>,
+                                        std::remove_cvref_t<decltype(bio::var_io::field_types_bcf_style<own>)>>;
+    ;
+
+    bio::format_input_handler<bio::vcf> handler{istr, bio::var_io::reader_options{}};
```

Comment:
I noticed that the `fasta_input_handler` has only a ctor from stream and the vcf one has only one from `stream, options`. Is this intended or will fasta get the same ctor too later on?

Reply:
1. both should have both now 🤔 
2. This is a little tricky. I wanted the formats to work without the "official" options object. This allows directly plugging the sam format in the seq_io reader for example. So each format has it's own options (described in the documentation), and then the options struct contains a subset or superset. There is no sync-ing mechanism for this other than the vigilant library developer 🛡️ 
3. They are not sync'ed. The parse_record_into function is a template. This allows reading all types that it can read.

Reply:
Ah ok, thanks for clarifying. This is tricky if you access the options though? For example if you access an option member in the sam_handler but this member isn't present when sequence_options are passed?

Reply:
That's why it checks for every option member that it understands individually with `if constexpr`